src/cargo/sources/git/known_hosts.rs

```diff
@@ -588,7 +590,18 @@ impl KnownHost {
         }
         for pattern in self.patterns.split(',') {
             let pattern = pattern.to_lowercase();
-            // FIXME: support * and ? wildcards
+            let is_glob = is_glob_pattern(&pattern);
+
+            if is_glob {
+                let regex_pattern =
+                    regex::escape(&pattern.replace("*", ",,,,").replace("?", ";;;;"))
+                        .replace(",,,,", ".*")
+                        .replace(";;;;", ".");
+                if let Ok(regex) = Regex::new(&regex_pattern) {
```

Comment:
Should we do anything on errors?

Reply:
Hmm, my thought is that if the pattern string could not be interpreted as glob pattern it's better to just pass glob matching.

Reply:
We could probably emit a `tracing::warn!` on failing cases.

Reply:
Sounds nice!